src/rust/engine/client/src/main.rs

```diff
@@ -96,10 +96,8 @@ fn try_execv_fallback_client(pants_server: OsString) -> Result<Infallible, i32>
 }
 
 fn execv_fallback_client(pants_server: OsString) -> Infallible {
-    if let Err(exit_code) = try_execv_fallback_client(pants_server) {
-        std::process::exit(exit_code);
-    }
-    unreachable!()
+    let Err(exit_code) = try_execv_fallback_client(pants_server);
```

Comment:
Rust is now able to infer that the `execv` will never return and thus the result will always be an `Err`.